src/main/java/org/jabref/gui/maintable/SmartConstrainedResizePolicy.java

```diff
@@ -22,75 +23,158 @@
+            LOGGER.trace("MaxWidth {}", maxWidth);
+            newWidth = Math.min(maxWidth, oldWidth + delta);
+        }
+        LOGGER.trace("Size: {} -> {}", oldWidth, newWidth);
+        if (oldWidth == newWidth) {
+            return Optional.empty();
         }
+        return Optional.of(newWidth);
     }
 
+    /**
+     * Resizes the table based on the content. The main driver is that if the content might fit into the table without horizontal scroll bar.
+     * In case the content fitted before the resize and will fit afterwards, the delta is distributed among the remaining columns - instead of just moving the columns right of the current column.
+     * In case the content does not fit anymore, a horizontal scroll bar is shown.
+     * In all cases the minimum/maximum width of each column is respected.
+     */
     private Boolean constrainedResize(TableView.ResizeFeatures<?> prop) {
         TableView<?> table = prop.getTable();
+        Double tableWidth = getContentWidth(table);
         List<? extends TableColumnBase<?, ?>> visibleLeafColumns = table.getVisibleLeafColumns();
-        return constrainedResize(prop,
-                false,
-                getContentWidth(table) - 2,
-                visibleLeafColumns);
+        Double requiredWidth = visibleLeafColumns.stream().mapToDouble(TableColumnBase::getWidth).sum();
+        Double delta = prop.getDelta();
+
+        TableColumn<?, ?> userChosenColumnToResize = prop.getColumn();
+
+        boolean columnsCanFitTable = requiredWidth + delta < tableWidth;
+        if (columnsCanFitTable) {
+            LOGGER.debug("User shrunk column in that way thus that window can contain all the columns");
+            if (requiredWidth >= tableWidth) {
+                LOGGER.debug("Before, the content did not fit. Rearrange everything.");
+                return rearrangeColumns(table);
+            }
+            LOGGER.debug("Everything already fit. We distribute the delta now.");
+
+            // Content does already fit
+            // We "just" need to readjust the column widths
+            determineNewWidth(userChosenColumnToResize, delta)
+                    .ifPresent(newWidth -> {
+                        double currentWidth = userChosenColumnToResize.getWidth();
+                        Double deltaToApply = newWidth - currentWidth;
+                        for (TableColumnBase<?, ?> col : visibleLeafColumns) {
+                            Double share = expansionShare.get(col);
+                            Double newSize;
+                            if (col.equals(prop.getColumn())) {
+                                // The column resized by the user gets the delta;
+                                newSize = newWidth;
+                            } else {
+                                // The columns not explicitly resized by the user get the negative delta
+                                newSize = col.getWidth() - share * deltaToApply;
```

Comment:
I think this doesn't result in a consistent table width. Imagine having two columns, with initial preferred widths of 70/30 and a total table width of 100. Now the user extends the first column by 10, so `deltaToApply = 10`. The first column then indeed has size 80 (as expected) but the second one has `newSize = 30 - 0.3 * 10 = 27` although it should be `20`.